### ISSUE TITLE
System.IO: Allow recursive file system enumeration to continue when directory link's target does not exist

### ISSUE THREAD

**Issue by fitdev** [dotnet/runtime#58612]
Perhaps this should be considered holistically as part of .Net 7's improved and more integrated support of file system links into various APIs, but at the least it would be nice to add:

```cs
public class EnumerationOptions {

  public bool IgnoreUnresolvableLinkTargets { get; set; } // when false preserves current (DotNet 6) behavior, i.e. throws DirectoryNotFoundException when target cannot be found

}
```

to prevent throwing of `DirectoryNotFoundException` when a symlink / junction is encountered during enumeration which has invalid (non-existing) target, as often it is valuable to continue enumeration regardless, plus it will parallel existing `IgnoreInaccessible` option.

**Comment by iSazonov** [comment]
I'd qualify this as a bug and re-use `EnumerationOptions.IgnoreInaccessible` to mask the exception.

/cc @carlossanlop @Jozkee

**Comment by FilipToth** [comment]
I agree with @iSazonov. The `IgnoreInaccessible` property has been designed with this goal in mind, making this obsolete.
